Library/Homebrew/cmd/update.rb

```diff
@@ -241,12 +241,12 @@ def pull!(options = {})
       return
     end
 
+    @initial_revision = read_current_revision
```

Comment:
Thanks! Without this being addressed, I believe running `brew update`, the report it generates, and the code that acts on this report will become mostly useless to me (as I'm almost always on some branch different from `master`).

My (possibly wrong) expectation is that `brew update` should fetch upstream changes and then only report and act on them, disregarding the stuff that might be going on in the personal branch that I'm currently on.